### PR TITLE
support special characters.

### PR DESCRIPTION
**Pull request by Genius-pig**
support special characters.

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/qp/plan/PhysicalPlanTest.java [hunk 1]
```diff
@@ -796,4 +796,11 @@ public void testLastPlanDataTypes() throws QueryProcessException {
       assertEquals(TSDataType.FLOAT, dt);
     }
   }
+
+  @Test
+  public void testSpecialCharacters() throws QueryProcessException {
+    String sqlStr1 = "create timeseries root.3e-3.-1.1/2.SNAPPY.RLE with datatype=FLOAT, encoding=RLE, compression=SNAPPY tags(tag1=v1, tag2=v2) attributes(attr1=v1, attr2=v2)";
+    PhysicalPlan plan1 = processor.parseSQLToPhysicalPlan(sqlStr1);
+    Assert.assertEquals(OperatorType.CREATE_TIMESERIES, plan1.getOperatorType());
```
Comment (Ring-k) [inline server/src/test/java/org/apache/iotdb/db/qp/plan/PhysicalPlanTest.java:804]: The path in the plan should be checked, not so serious though.

File: server/src/main/antlr4/org/apache/iotdb/db/qp/strategy/SqlBase.g4 [hunk 2]
```diff
@@ -37,7 +37,7 @@ statement
     | CREATE INDEX ON fullPath USING function=ID indexWithClause? whereClause? #createIndex //not support yet
     | DROP INDEX function=ID ON fullPath #dropIndex //not support yet
     | MERGE #merge
-    | FLUSH prefixPath? (COMMA prefixPath)* (ID)?#flush //ID is true or false
+    | FLUSH prefixPath? (COMMA prefixPath)* (booleanClause)?#flush //ID is true or false
```
Comment (Ring-k) [inline server/src/main/antlr4/org/apache/iotdb/db/qp/strategy/SqlBase.g4:40]: The comment should be updated.

File: docs/UserGuide/5-Operation Manual/4-SQL Reference.md [hunk 3]
```diff
@@ -32,8 +32,6 @@ All of these statements are write in IoTDB's own syntax, for details about the s
 
 ## Keywords
 
-Please not use these keywords as identifiers.
```
Comment (Ring-k) [inline docs/UserGuide/5-Operation Manual/4-SQL Reference.md:35]: It seems that this hint still works.

File: docs/zh/UserGuide/5-Operation Manual/4-SQL Reference.md [hunk 4]
```diff
@@ -23,8 +23,6 @@
 
 ## 关键字
 
-不要使用这些关键字作为标识符。
```
Comment (Ring-k) [inline docs/zh/UserGuide/5-Operation Manual/4-SQL Reference.md:26]: It seems that this hint still works.

File: server/src/test/java/org/apache/iotdb/db/qp/plan/PhysicalPlanTest.java (hunk 1 above)
Comment (Genius-pig) [reply server/src/test/java/org/apache/iotdb/db/qp/plan/PhysicalPlanTest.java:804]: I think it's OK.

**Pull request merged**